medtronic/src/main/java/info/nightscout/androidaps/plugins/pump/medtronic/defs/MedtronicCommandType.kt

```diff
@@ -0,0 +1,266 @@
+    SetBasalProfileB(0x31, "Set Profile B", MedtronicDeviceType.Medtronic_512andHigher, MinimedCommandParameterType.NoParameters,  //
+        64, 3, 192, R.string.medtronic_cmd_desc_set_basal_profile),  // 49
+
+    // 515
+    PumpStatus(206, "Pump Status", MedtronicDeviceType.Medtronic_515andHigher, MinimedCommandParameterType.NoParameters),  // PumpConfiguration
+    Settings(192, "Configuration", MedtronicDeviceType.Medtronic_515andHigher, MinimedCommandParameterType.NoParameters,  //
+        64, 1, 21, R.string.medtronic_cmd_desc_get_settings),  //
+
+    // 522
+    SensorSettings_522(153, "Sensor Configuration", MedtronicDeviceType.Medtronic_522andHigher, MinimedCommandParameterType.NoParameters),  //
+    GlucoseHistory(154, "Glucose History", MedtronicDeviceType.Medtronic_522andHigher, MinimedCommandParameterType.SubCommands, 1024, 32, 0, null),  //
+
+    // 523
+    SensorSettings(207, "Sensor Configuration", MedtronicDeviceType.Medtronic_523andHigher, MinimedCommandParameterType.NoParameters),  //
+
+    // 553
+    // 554
+    // var MESSAGES = {
+    // READ_TIME : 0x70,
+    // READ_BATTERY_STATUS: 0x72,
+    // READ_HISTORY : 0x80,
+    // READ_CARB_RATIOS : 0x8A,
+    // READ_INSULIN_SENSITIVITIES: 0x8B,
+    // READ_MODEL : 0x8D,
+    // READ_PROFILE_STD : 0x92,
+    // READ_PROFILE_A : 0x93,
+    // READ_PROFILE_B : 0x94,
+    // READ_CBG_HISTORY: 0x9A,
+    // READ_ISIG_HISTORY: 0x9B,
+    // READ_CURRENT_PAGE : 0x9D,
+    // READ_BG_TARGETS : 0x9F,
+    // READ_SETTINGS : 0xC0, 192
+    // READ_CURRENT_CBG_PAGE : 0xCD
+    // };
+    // Fake Commands
+    CancelTBR;
+
+    companion object {
+
+        var mapByCode: MutableMap<Byte, MedtronicCommandType> = HashMap()
+
+        // private fun getDeviceTypesArray(vararg types: MedtronicDeviceType): HashMap<MedtronicDeviceType, String?> {
+        //     val hashMap = HashMap<MedtronicDeviceType, String?>()
+        //     for (type in types) {
+        //         hashMap[type] = null
+        //     }
+        //     return hashMap
+        // }
+
+        fun getByCode(code: Byte): MedtronicCommandType? {
+            return if (mapByCode.containsKey(code)) {
+                mapByCode[code]
+            } else {
+                InvalidCommand
+            }
+        }
+
+        fun constructMessageBody(messageType: MedtronicCommandType?, bodyData: ByteArray): MessageBody {
+            return when (messageType) {
+                CommandACK -> PumpAckMessageBody(bodyData)
+                else       -> UnknownMessageBody(bodyData)
+            }
+        }
+
+        @JvmStatic
+        fun getSettings(medtronicPumpModel: MedtronicDeviceType): MedtronicCommandType {
+            return if (isSameDevice(medtronicPumpModel, MedtronicDeviceType.Medtronic_512_712))
+                Settings_512
+            else
+                Settings
+        }
+
+        init {
+            RFPowerOn.maxAllowedTime = 17000
```

Comment:
This is dynamically modifying fields of an Enum.
Here I'd add it as constructor parameter (with default)

Reply:
Yeah I refactored that class now...